src/web_interface/rest/rest_firmware.py

```diff
@@ -80,26 +80,31 @@ def _update_analysis(self, uid, update):
         return success_message({}, self.URL, dict(uid=uid, update=update))
 
     def _get_without_uid(self):
-        paging, success = get_paging(request.args)
-        if not success:
-            return error_message(paging, self.URL, request_data=request.args)
-        offset, limit = paging
-
         try:
-            recursive = get_recursive(request.args)
-            query = get_query(request.args)
+            query, recursive, inverse, (offset, limit) = self._get_parameters_from_request(request.args)
         except ValueError as value_error:
-            return error_message(str(value_error), self.URL, request_data=dict(query=request.args.get('query'), recursive=request.args.get('recursive')))
-        if recursive and not query:
-            return error_message('recursive search is only permissible with non-empty query', self.URL, request_data=dict(query=request.args.get('query'), recursive=request.args.get('recursive')))
+            request_data = {k: request.args.get(k) for k in ['query', 'limit', 'offset', 'recursive', 'inverse']}
+            return error_message(str(value_error), self.URL, request_data=request_data)
 
+        parameters = dict(offset=offset, limit=limit, query=query, recursive=recursive, inverse=inverse)
         try:
             with ConnectTo(FrontEndDbInterface, self.config) as connection:
-                uids = connection.rest_get_firmware_uids(offset=offset, limit=limit, query=query, recursive=recursive)
+                uids = connection.rest_get_firmware_uids(**parameters)
+            return success_message(dict(uids=uids), self.URL, parameters)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
See above. Maybe we can replace the broad except

Reply:
changed to `PyMongoError`